ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/LangPack/tasks/ja.yml

```diff
@@ -0,0 +1,77 @@
+---
+#################
+# Language Pack #
+#################
+
+- debug:
+    msg: "OS: {{ ansible_facts['distribution'] }}"
+  tags: langpack
+
+- name: Set Windows 2016 Packages
+  set_fact:
+    JP_cab: /Vendor_Files/langpack/Win2016/x64fre_Server_ja-jp_lp.cab
+  when: ansible_facts['distribution'] is match('^Microsoft Windows Server 2016')
+  tags: langpack
+
+- name: Set Windows 2012R2 Packages
+  set_fact:
+    JP_cab: /Vendor_Files/langpack/Win2012R2/jp/lp.cab
+  when: ansible_facts['distribution'] is match('^Microsoft Windows Server 2012 R2')
+  tags: langpack
+
+- name: Set Windows 2012 Packages
+  set_fact:
+    JP_cab: /Vendor_Files/langpack/Win2012/jp/lp.cab
+  when:
+    - ansible_facts['distribution'] is match('^Microsoft Windows Server 2012')
+    - ansible_facts['distribution'] is not match('^Microsoft Windows Server 2012 R2')
+  tags: langpack
+
+- name: Set Windows 10 Packages
+  set_fact:
+    JP_cab: /Vendor_Files/langpack/Win10/Microsoft-Windows-Client-Language-Pack_x64_ja-jp.cab
+  when: ansible_facts['distribution'] is match('^Microsoft Windows 10')
+  tags: langpack
+
+- name: Test if language pack ja is already installed
+  win_shell: |
+      DISM.exe /Online /Get-Intl /English |
+      Select-String -SimpleMatch 'Installed language(s)'|
+      ForEach-Object { if($_ -match 'ja-JP'){ $Matches[0] }}
+  register: jp_installed
+  tags: langpack
+
+- name: install language pack ja
+  win_shell: "Dism /online /Add-Package /PackagePath:{{ JP_cab }}"
+  when: (jp_installed.stdout | length == 0) and JP_cab is defined
```

Comment:
If JP_cab is undefined at this point, the playbook fails, example my windows8 install.  Please add a when JP_cab is defined here.

Reply:
Thanks you for the comment, but I'm afraid the last condition is the one.

Reply:
I have no idea how I missed that!  Sorry.